pyodide_http/_streaming.py

```diff
@@ -0,0 +1,231 @@
+                return;
+            }
+        }
+    }
+    catch (error) {
+            console.log("Request exception:",error);
+            let errorBytes=encoder.encode(errorBytes);
+            let written = errorBytes.length;
+            byteBuffer.set(errorBytes);
+            intBuffer[1] = written;
+            Atomics.store(intBuffer, 0, ERROR_EXCEPTION);
+            Atomics.notify(intBuffer,0);
+        }
+    }
+
+);
+
+"""
+
+
+def _obj_from_dict(dict_val: dict) -> any:
+    return to_js(dict_val, dict_converter=js.Object.fromEntries)
+
+
+class _ReadStream(io.RawIOBase):
+    def __init__(self, int_buffer, byte_buffer):
+        self.int_buffer = int_buffer
+        self.byte_buffer = byte_buffer
+        self.read_pos = 0
+        self.read_len = 0
+
+    def readable(self) -> bool:
+        return True
+
+    def writeable(self) -> bool:
+        return False
+
+    def seekable(self) -> bool:
+        return False
+
+    def readinto(self, byte_obj) -> bool:
+        if not self.int_buffer:
+            return 0
+        if self.read_len == 0:
+            # wait for the worker to send something
+            js.Atomics.store(self.int_buffer, 0, 0)
+            js.Atomics.notify(self.int_buffer, 0)
+            js.Atomics.wait(self.int_buffer, 0, 0, 5000)
+            data_len = self.int_buffer[0]
+            if data_len > 0:
+                self.read_len = data_len
+                self.read_pos = 0
+            else:
+                # EOF, free the buffers and return zero
+                self.read_len = 0
+                self.read_pos = 0
+                self.int_buffer = None
+                self.byte_buffer = None
+                return 0
+        # copy from int32array to python bytes
+        ret_length = min(self.read_len, len(byte_obj))
+        self.byte_buffer.subarray(
+            self.read_pos, self.read_pos+ret_length).assign_to(byte_obj[0:ret_length])
+        self.read_len -= ret_length
+        self.read_pos += ret_length
+        return ret_length
+
+
+class _Streaming_Fetcher:
+    def __init__(self):
+        # make web-worker and data buffer on startup
+        dataBlob = js.Blob.new([_STREAMING_WORKER_CODE], _obj_from_dict(
+            {"type": 'application/javascript'}))
+        dataURL = js.URL.createObjectURL(dataBlob)
+        self._worker = js.Worker.new(dataURL)
+
+    def send(self, request):
+        from ._core import Response
+        headers = _obj_from_dict(request.headers)
+        body = request.body
+        fetch_data = _obj_from_dict(
+            {"headers": headers, "body": body, "method": request.method})
+        # start the request off in the worker
+
+        shared_buffer = js.SharedArrayBuffer.new(1048576)
+        int_buffer = js.Int32Array.new(shared_buffer)
+        byte_buffer = js.Uint8Array.new(shared_buffer, 8)
+
+        js.Atomics.store(int_buffer, 0, 0)
+        js.Atomics.notify(int_buffer, 0)
+        absolute_url = js.URL.new(request.url, js.location).href
```

Comment:
url params are missing, right?

Reply:
the url params get added into the url itself in send, so they're already in this.